posthog.go

```diff
@@ -37,17 +37,20 @@ type Client interface {
 	Enqueue(Message) error
 	//
 	// Method returns if a feature flag is on for a given user based on their distinct ID
-	IsFeatureEnabled(string, string, bool) (bool, error)
+	IsFeatureEnabled(string, string, bool, Groups, Properties, map[string]Properties) (bool, error)
```

Comment:
We can use [variadic functions](https://gobyexample.com/variadic-functions) here but we'll lose type enforcement